### PR TITLE
chore: Migrate WebDAV `files_external` test from Drone to GitHub

### PR DESCRIPTION
**Pull request by susnux**
* For #41187 

## Summary
Migrates the Apache WebDAV files_external from Drone to GitHub.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: autotest-external.sh [hunk 1]
```diff
@@ -31,8 +31,13 @@ function print_syntax {
 }
 
 if ! [ -x "$PHPUNIT" ]; then
-	echo "phpunit executable not found, please install phpunit version >= 4.8" >&2
-	exit 3
+	echo "phpunit executable not found, trying local one from build/integration" >&2
+	if [ -x "$PWD/build/integration/vendor/phpunit/phpunit/phpunit" ]; then
+		PHPUNIT="$PWD/build/integration/vendor/phpunit/phpunit/phpunit"
+	else
+		echo "phpunit executable not found, please install phpunit version >= 4.8" >&2
```
Comment (nickvergessen) [inline autotest-external.sh:38]: Bump this
```suggestion
		echo "phpunit executable not found, please install phpunit version >= 9.6" >&2
```

4.8 was for PHP 5.x support...

Comment (susnux) [reply autotest-external.sh:38]: Not changed but yes can do so

**Comment by susnux** [comment]
@nickvergessen I think you need to sign the drone config if approved.

**Comment by nickvergessen** [comment]
> I think you need to sign the drone config if approved.

I did, you force pushed over

**Review by nickvergessen — APPROVED** [review]

**Review by come-nc — APPROVED** [review]
Good apart from the coverage thing

File: .github/workflows/files-external-webdav.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    needs: changes
+
+    if: ${{ github.repository_owner != 'nextcloud-gmbh' && needs.changes.outputs.src != 'false' }}
+
+    strategy:
+      matrix:
+        php-versions: ['8.0', '8.1', '8.2', '8.3']
+        include:
+          - php-versions: '8.3'
+            coverage: true
```
[the remark this answers is not in the document]
Comment (susnux) [reply .github/workflows/files-external-webdav.yml]: Same for S3 currently, but I applied the same patch like on nodb now.

File: autotest-external.sh (hunk 1 above)
Comment (susnux) [reply autotest-external.sh]: @nickvergessen this breaks Drone as drone uses PHPUnit 9.4.4 not 9.6.0

File: autotest-external.sh [hunk 3]
```diff
@@ -31,16 +31,21 @@ function print_syntax {
...
+	fi
 fi
 
 PHPUNIT_VERSION=$("$PHPUNIT" --version | cut -d" " -f2)
 PHPUNIT_MAJOR_VERSION=$(echo $PHPUNIT_VERSION | cut -d"." -f1)
 PHPUNIT_MINOR_VERSION=$(echo $PHPUNIT_VERSION | cut -d"." -f2)
 
 if ! [ $PHPUNIT_MAJOR_VERSION -gt 9 -o \( $PHPUNIT_MAJOR_VERSION -eq 9 -a $PHPUNIT_MINOR_VERSION -ge 0 \) ]; then
-	echo "phpunit version >= 9.0 required. Version found: $PHPUNIT_VERSION" >&2
+	echo "phpunit version >= 9.6 required. Version found: $PHPUNIT_VERSION" >&2
```
Comment (nickvergessen) [inline autotest-external.sh:48]:
```suggestion
	echo "phpunit version >= 9.0 required. Version found: $PHPUNIT_VERSION" >&2
```

Or you adjust the if condition above and the install hint in line 38

**Review by nickvergessen — APPROVED** [review]

File: autotest-external.sh (hunk 3 above)
Comment (susnux) [reply autotest-external.sh:48]: will not work with drone that only provides 9.4.4

**Pull request merged**